src/renderer.cpp

```diff
@@ -312,6 +334,9 @@ void Renderer::resetState()
     doModulateAlpha();
     doTextureFiltering();
     isExternalTexture();
+    setCastShadow();
+
+    internalState.lightingSettings[0] = enable_lighting;

```

Comment:
Technically this flag should be set only once in a lifecycle, or can it be resetted through the debug devtool?

Reply:
Yes, lighting can be enabled/disabled from the debug devtools.

Reply:
Find a way to interactively reset it through a class method instead of changing a global variable which is then set on each pass. Thanks.